grayskull/pypi/pypi.py

```diff
@@ -59,31 +59,38 @@ def __init__(
         self["build"]["script"] = "<{ PYTHON }} -m pip install . -vv"
 
     @staticmethod
-    def _download_sdist_pkg(sdist_url: str, dest: str):
+    def _download_sdist_pkg(sdist_url: str, name: str, dest: str):
         """Download the sdist package
 
         :param sdist_url: sdist url
         :param dest: Folder were the method will download the sdist
         """
-        name = sdist_url.split("/")[-1]
+        #name = sdist_url.split("/")[-1]
         print_msg(
             f"{Fore.GREEN}Starting the download of the sdist package"
             f" {Fore.BLUE}{Style.BRIGHT}{name}"
         )
         log.debug(f"Downloading {name} sdist - {sdist_url}")
         response = requests.get(sdist_url, allow_redirects=True, stream=True, timeout=5)
-        total_size = int(response.headers["Content-length"])
-
-        with manage_progressbar(max_value=total_size, prefix=f"{name} ") as bar, open(
-            dest, "wb"
-        ) as pkg_file:
-            progress_val = 0
-            chunk_size = 512
-            for chunk_data in response.iter_content(chunk_size=chunk_size):
-                if chunk_data:
-                    pkg_file.write(chunk_data)
-                    progress_val += chunk_size
-                    bar.update(min(progress_val, total_size))
+        print(response.headers)
+        try:
+            total_size = int(response.headers["Content-length"])
+            with manage_progressbar(max_value=total_size, prefix=f"{name} ") as bar, open(dest, "wb") as pkg_file:
+                progress_val = 0
+                chunk_size = 512
+                for chunk_data in response.iter_content(chunk_size=chunk_size):
+                    if chunk_data:
+                        pkg_file.write(chunk_data)
+                        progress_val += chunk_size
+                        bar.update(min(progress_val, total_size))
+        except KeyError as error:
```

Comment:
I don't think we should try/except here. The `KeyError` comes from the response headers missing `Content-length`, which is only used to instantiate the progress bar. Providing a default (like `0`, which many progress bars use as a placeholder for "we don't know how large this is, let's just spin around until we are done") might work. Otherwise, the current except clause will cause us to download the whole thing in memory, which can be a huge lot if the repo is particularly large.

Reply:
Jaime, would you please explain this  a little more?

Reply:
You needed try/except here to work around the fact that some requests to GitHub return a response where the headers do not contain a `Content-length` entry. This content length is used only to create a nice progress bar, so we can just return a number that means "hey, we don't really know how big this is, but tell the user we are doing something anyway - we can tell you when it's done!".

The problem with this particular block of code is that the fallback part in the except clause does not really download the file in the same way (progressively, chunk by chunk, writing to disk, avoids having to store the whole thing in memory before writing in to disk.